.github/workflows/scheduled_aas_deploy.yml

```diff
@@ -0,0 +1,101 @@
+name: Scheduled AAS Deployment
+
+on:
+  schedule:
+    - cron: '30 21 * * FRI'  # Every Friday at 9:30 PM UTC
+  workflow_dispatch:
+    inputs:
+      force_deploy:
+        description: '[TESTING] Force deployment regardless of code freeze status'
+        type: boolean
+        default: false
+      override_week:
+        description: '[TESTING] Override week number (empty = use current week, even number = deploy)'
+        required: false
+        type: string
+
+jobs:
+  check_freeze_and_deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      actions: read
+      issues: write
+    
+    steps:
+      - name: Calculate week number
+        id: week
+        run: |
+          if [ ! -z "${{ inputs.override_week }}" ]; then
+            echo "Using override week: ${{ inputs.override_week }}"
+            WEEK_NUM="${{ inputs.override_week }}"
+          else
+            WEEK_NUM=$(date +%V)
+          fi
+          WEEK_NUM=$((10#$WEEK_NUM)) # Convert to decimal to handle leading zeros
+          echo "Week number being used: $WEEK_NUM"
+          echo "week_number=$WEEK_NUM" >> $GITHUB_OUTPUT
+
+      - name: Check if we should run
+        id: should_run
+        run: |
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" && "${{ inputs.force_deploy }}" == "true" ]]; then
+            echo "should_proceed=true" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          # Check if week number is even (proceeding with deployment)
+          if (( ${{ steps.week.outputs.week_number }} % 2 == 0 )); then
+            echo "should_proceed=true" >> $GITHUB_OUTPUT
+          else
+            echo "should_proceed=false" >> $GITHUB_OUTPUT
+          fi
+
+      # Check skip milestone
+      - uses: octokit/request-action@786351db496fa66730d8faa09ef279108da175a3 # v2.x
+        if: steps.should_run.outputs.should_proceed == 'true'
+        name: 'Check Skip Milestone'
+        id: skip_milestone
+        with:
+          route: GET /repos/{owner}/{repo}/milestones/183 # Skip Scheduled Code Freeze milestone ID
+          owner: ${{ github.repository_owner }}
+          repo: ${{ github.event.repository.name }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Process skip state
+        if: steps.should_run.outputs.should_proceed == 'true'
+        id: check_skip
+        run: |
+          state=$(echo '${{ steps.skip_milestone.outputs.data }}' | jq -r '.state')
```

Comment:
I assume the skip milestone is a safety check way of skipping the freeze and deploy? Like, if we _know_ we don't want to freeze for some reason? 

Presumably we wouldn't expect to need to use this often, and worst case you could run the end code freeze workflow too, right? We would manually open the milestone to enable it right?

Reply:
Correct!

The idea that I have is:

- Thursday we have an automated slack message about upcoming freeze
- If we don't want to freeze we run that to skip it without needing to do a code change on them
- Code freeze would be skipped
- Deploy would be skipped and then the deploy closes the skip milestone

I'm working on a doc / diagram to show the flow of everything